src/verify.c

```diff
@@ -467,68 +467,79 @@ static void add_missing_files(struct manifest *official_manifest)
-	iter = list_head(official_manifest->files);
-	while (iter) {
-		struct file *file;
-		char *fullname;
+	// Note: boot files not marked as deleted are candidates for verify/fix
+	if (file->is_deleted ||
+	    ignore(file)) {
+		return;
+	}
 
-		file = iter->data;
-		iter = iter->next;
+	file_checked_count++;
 
-		// Note: boot files not marked as deleted are candidates for verify/fix
-		if (file->is_deleted ||
-		    ignore(file)) {
-			continue;
-		}
+	// do_not_update set by earlier check, so account as checked
+	if (file->do_not_update) {
+		return;
+	}
 
-		file_checked_count++;
+	/* compare the hash and report mismatch */
+	fullname = mk_full_filename(path_prefix, file->filename);
+	if (fullname == NULL) {
+		abort();
+	}
+	if (verify_file(file, fullname)) {
+		goto end;
+	}
+	file_mismatch_count++;
+	/* Log to stdout, so we can post-process it */
+	printf("Hash mismatch for file: %s\n", fullname);
 
-		// do_not_update set by earlier check, so account as checked
-		if (file->do_not_update) {
-			continue;
-		}
 
-		/* compare the hash and report mismatch */
-		fullname = mk_full_filename(path_prefix, file->filename);
-		if (fullname == NULL) {
-			abort();
-		}
-		if (verify_file(file, fullname)) {
-			free(fullname);
-			continue;
-		} else {
-			file_mismatch_count++;
-			/* Log to stdout, so we can post-process it */
-			printf("Hash mismatch for file: %s\n", fullname);
-		}
+	/* if not repairing, we're done */
+	if (!repair) {
+		goto end;
+	}
 
-		/* if not repairing, we're done */
-		if (!repair) {
-			free(fullname);
-			continue;
-		}
+	/* install the new file (on miscompare + fix) */
+	ret = do_staging(file, official_manifest);
+	if (ret == 0) {
+		rename_staged_file_to_final(file);
+	}
 
-		/* install the new file (on miscompare + fix) */
-		ret = do_staging(file, official_manifest);
-		if (ret == 0) {
-			rename_staged_file_to_final(file);
-		}
+	/* at the end of all this, verify the hash again to judge success */
+	if (verify_file(file, fullname)) {
+		file_fixed_count++;
+		printf("\tfixed\n");
+	} else {
+		file_not_fixed_count++;
+		printf("\tnot fixed\n");
+	}
+end:
+	free(fullname);
+}
 
-		/* at the end of all this, verify the hash again to judge success */
-		if (verify_file(file, fullname)) {
-			file_fixed_count++;
-			printf("\tfixed\n");
-		} else {
-			file_not_fixed_count++;
-			printf("\tnot fixed\n");
-		}
-		free(fullname);
+static void deal_with_hash_mismatches(struct manifest *official_manifest, bool repair)
+{
+	struct list *iter;
+	int complete=0;
+	int list_length;
+
+	/* for each expected and present file which hash-mismatches vs
+	 * the manifest, replace the file */
+	iter = list_head(official_manifest->files);
+	list_length = list_len(iter);
+	
+	while (iter) {
+		struct file *file;
+		file = iter->data;
+		iter = iter->next;
+		complete++;
+
+		check_and_fix_one(file, official_manifest, repair);
+		print_progress(complete, list_length);
```

Comment:
You'll need to print a newline after this to prevent the following output:
```
Verifying files
        ...100%Inspected 295658 files
Verify successful
```

Reply:
Added below at line 543